.github/pull_request_template.md

```diff
@@ -0,0 +1,19 @@
+<-- Start the PR description with some context for the change. -->
+
+
+<-- Make sure to delete the markdown comments and the below sections when squash merging -->
+### Issue describing the changes in this PR
+
+resolves #issue_for_this_pr
+
+### Pull request checklist
+
+* [ ] My changes **do not** require documentation changes
+    * [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
+* [ ] My changes **should not** be added to the release notes for the next release
+    * [ ] Otherwise: I've added my notes to `release_notes.md`
+* [ ] My changes **do not** need to be backported to a previous version
```

Comment:
What does this mean? When would I need to do this?

Reply:
Yeah, this is a great question. I was debating on leaving this off, as its hopefully a very rare thing.

For a while after releasing Durable Functions v2.x, we had to still backport fixes to Durable Functions 1.x, as customers were still struggling to update to v2.x due to the breaking changes. It was even more of a problem for Durable JS customers, because they didn't have an easy way to move to v2.x until extension bundles were solidified. Durable JS customers can now move to v2.x with a simple host.json change, so it's a bit less of a concern now.

Our last Durable 1.x release was v1.8.6 released ~6 months ago. I wouldn't be opposed to us updating the underlying version of DurableTask.AzureStorage every few motnsh to get some extra reliability fixes in for JavaScript customers who never moved off of extension bundles v1, but most PRs should be v2 only now.